### ISSUE TITLE
azurerm_windows_virtual_machine_scale_set perpetual diff with extension protected_settings

### ISSUE THREAD

**Issue by hbuckle** [hashicorp/terraform-provider-azurerm#11820]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.14.11
+ provider registry.terraform.io/hashicorp/azurerm v2.60.0

### Affected Resource(s)

* `azurerm_windows_virtual_machine_scale_set `

### Terraform Configuration Files

```hcl

resource "azurerm_windows_virtual_machine_scale_set" "node_type" {
  name                     = "extension"
  resource_group_name      = azurerm_resource_group.extension.name
  location                 = azurerm_resource_group.extension.location
  admin_username           = "azureuser"
  admin_password           = var.admin_password
  license_type             = "None"
  custom_data              = null
  upgrade_mode             = "Automatic"
  computer_name_prefix     = "prefix"
  instances                = 0
  sku                      = "Standard_E8s_v3"
  priority                 = "Regular"
  overprovision            = false
  single_placement_group   = true
  enable_automatic_updates = false
  provision_vm_agent       = true

  source_image_reference {
    publisher = "MicrosoftWindowsServer"
    offer     = "WindowsServer"
    sku       = "datacenter-core-20h2-with-containers-smalldisk-g2"
    version   = "Latest"
  }

  automatic_os_upgrade_policy {
    disable_automatic_rollback  = false
    enable_automatic_os_upgrade = false
  }

  os_disk {
    caching              = "ReadWrite"
    storage_account_type = "Premium_LRS"
  }

  network_interface {
    name    = "primary"
    primary = true

    ip_configuration {
      name      = "primary"
      primary   = true
      subnet_id = var.subnet_id
    }
  }

  extension {
    name                       = "init"
    publisher                  = "Microsoft.Compute"
    type                       = "CustomScriptExtension"
    type_handler_version       = "1.10"
    auto_upgrade_minor_version = true
    protected_settings = jsonencode({
      commandToExecute = "powershell -NonInteractive -NoProfile -Command exit 0"
    })
  }
}

```
### Actual Behaviour

Now that inline extensions are a set if you add an extension with protected_settings you get a perpetual diff

```
Terraform will perform the following actions:

  # azurerm_windows_virtual_machine_scale_set.node_type will be updated in-place
  ~ resource "azurerm_windows_virtual_machine_scale_set" "node_type" {
        name                                              = "extension"
        tags                                              = {}
        # (22 unchanged attributes hidden)



      - extension {
          - auto_upgrade_minor_version = true -> null
          - name                       = "init" -> null
          - provision_after_extensions = [] -> null
          - publisher                  = "Microsoft.Compute" -> null
          - type                       = "CustomScriptExtension" -> null
          - type_handler_version       = "1.10" -> null
        }
      + extension {
          + auto_upgrade_minor_version = true
          + name                       = "init"
          + protected_settings         = (sensitive value)
          + provision_after_extensions = []
          + publisher                  = "Microsoft.Compute"
          + type                       = "CustomScriptExtension"
          + type_handler_version       = "1.10"
        }



        # (5 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

### Steps to Reproduce

1. `terraform apply`
1. `terraform plan`

### Important Factoids

Looking at the old resource there was a custom `Set` function - https://github.com/terraform-providers/terraform-provider-azurerm/blob/84d710e21fca7abcf36f513f1c278c3151721fb1/azurerm/internal/services/compute/virtual_machine_scale_set_resource.go#L774

I guess this needs to be added to the new resource as well

**Comment by magodo** [comment]
@hbuckle This is issue relates to #11425. After changing the `extension` from List to Set, the code below doesn't work any more:
 https://github.com/terraform-providers/terraform-provider-azurerm/blob/3dbd4a80597e8ff362ea4e051252cdd91af97874/azurerm/internal/services/compute/virtual_machine_scale_set.go#L1671-L1676

**Comment by hbuckle** [comment]
@magodo doesn't it just need the custom `Set` function adding from the old resource, that ignores protected settings when calculating the hashcode?

**Comment by mivica** [comment]
Hi, I am also observing this issue using AzureRM v2.68.0. Can anyone tell when will this get fixed?

**Comment by Va55ago** [comment]
Same here using azurerm v2.68.0.

Any extension with protected settings causes the VMSS to be updated on every apply, even if nothing has changed.

**Comment by hbuckle** [comment]
Just needs the linked pull request, but it seems to be impossible to get a review on it 😢

**Comment by rem-aj** [comment]
Hi @mbfrahry will the fix be pushed out in the 2.79 release?

**Comment by magodo** [comment]
@mivica I've just tested locally using v2.81.0. I apply with the config above and run a `terraform plan` afterwards, it shows no plan diff. Would you please kindly provide a detailed repro step so that I can dig into?

**Comment by rem-aj** [comment]
Hi @katbyte the bot mentioned this has been released in version 2.83, however there's an update from you it's moved to 2.84 and the issue is still "open", so I assume we have to wait for .84 then? Seems the bot note is inaccurate.

**Comment by Va55ago** [comment]
I've tried this out today (using ```terraform v1.0.8``` & ```azurerm v2.83.0```) and it successfully recognised that there were no changes to my VMSS.

Based on that, I think the bot is right and this was released in v2.83.0.

Thanks @magodo  :)

**Comment by tombuildsstuff** [comment]
fixed via #13440
